worker/src/sql.rs

```diff
@@ -0,0 +1,216 @@
+    fn from(val: SqlStorageValue) -> Self {
+        match val {
+            SqlStorageValue::Null => JsValue::NULL,
+            SqlStorageValue::Boolean(b) => JsValue::from(b),
+            SqlStorageValue::Integer(i) => JsValue::from(i),
+            SqlStorageValue::Float(f) => JsValue::from(f),
+            SqlStorageValue::String(s) => JsValue::from(s),
+            SqlStorageValue::Blob(bytes) => {
+                // Convert Vec<u8> to Uint8Array for JavaScript
+                let array = js_sys::Uint8Array::new_with_length(bytes.len() as u32);
+                array.copy_from(&bytes);
+                array.into()
+            }
+        }
+    }
+}
+
+/// Wrapper around the Workers `SqlStorage` interface exposed at `ctx.storage.sql`.
+///
+/// The API is intentionally minimal for now – additional helper utilities can be built on top
+/// as needed.
+#[derive(Clone, Debug)]
+pub struct SqlStorage {
+    inner: SqlStorageSys,
+}
+
+unsafe impl Send for SqlStorage {}
+unsafe impl Sync for SqlStorage {}
+
+impl SqlStorage {
+    pub(crate) fn new(inner: SqlStorageSys) -> Self {
+        Self { inner }
+    }
+
+    /// Size of the underlying SQLite database in bytes.
+    pub fn database_size(&self) -> usize {
+        self.inner.database_size() as usize
+    }
+
+    /// Execute a SQL query and return a cursor over the results.
+    ///
+    /// `bindings` correspond to positional `?` placeholders in the query.
+    /// Accepts `SqlStorageValue` for type-safe parameter binding.
+    pub fn exec(
+        &self,
+        query: &str,
+        bindings: impl Into<Option<Vec<SqlStorageValue>>>,
+    ) -> Result<SqlCursor> {
+        let array = Array::new();
+        if let Some(bindings) = bindings.into() {
+            for v in bindings {
+                array.push(&v.into());
+            }
+        }
+        let cursor = self.inner.exec(query, array).map_err(Error::from)?;
+        Ok(SqlCursor { inner: cursor })
+    }
+
+    /// Execute a SQL query with raw JavaScript values.
+    ///
+    /// This method provides direct access to `JsValue` parameters for advanced use cases
+    /// where you need more control over the JavaScript conversion.
+    pub fn exec_raw(
+        &self,
+        query: &str,
+        bindings: impl Into<Option<Vec<JsValue>>>,
+    ) -> Result<SqlCursor> {
+        let array = Array::new();
+        if let Some(bindings) = bindings.into() {
+            for v in bindings {
+                array.push(&v);
+            }
+        }
+        let cursor = self.inner.exec(query, array).map_err(Error::from)?;
+        Ok(SqlCursor { inner: cursor })
+    }
+}
+
+impl AsRef<JsValue> for SqlStorage {
+    fn as_ref(&self) -> &JsValue {
+        &self.inner
+    }
+}
+
+/// A cursor returned from `SqlStorage::exec`.
+#[derive(Clone, Debug)]
+pub struct SqlCursor {
+    inner: SqlStorageCursorSys,
+}
+
+unsafe impl Send for SqlCursor {}
+unsafe impl Sync for SqlCursor {}
+
+impl SqlCursor {
```

Comment:
Seems this is missing:
* Iterator
* resusedCachedQueryForTest
* raw
* next

Any reason not to consider adding any of these as well?

Reply:
added raw,next, and Iterator impl. not 100% sure I got those right. 

left a todo and ref for resusedCachedQueryForTest